frame/contracts/src/lib.rs

```diff
@@ -1172,6 +1215,12 @@ impl<T: Config> Pallet<T> {
 			debug_message: debug_message.as_mut(),
 		};
 		let output = InstantiateInput::<T> { code, salt }.run_guarded(common);
+		// collect events if CollectEvents is UnsafeCollect
+		let events = if matches!(collect_events, CollectEvents::UnsafeCollect) {
```

Comment:
Nit: With `#[derive(PartialEq)]` on the enum, `if collect_events == CollectEvents::UnsafeCollect {}` would be possible